### PR TITLE
Convert dhtest, dsatest, cipherbytes_test

### PR DESCRIPTION
**Pull request by richsalz**

### CODE REVIEW

File: test/dhtest.c [hunk 1]
```diff
@@ -580,19 +491,23 @@ static int run_rfc5114_tests(void)
     BN_free(priv_key);
     OPENSSL_free(Z1);
     OPENSSL_free(Z2);
-
-    fprintf(stderr, "Initialisation error RFC5114 set %d\n", i + 1);
-    ERR_print_errors_fp(stderr);
+    TEST_error("Initialisation error RFC5114 set %d\n", i + 1);
```
Comment (levitte) [inline test/dhtest.c]: I think `ERR_print_errors_fp(stderr)` should stay, so we get the errors related to this message *with* this message, and not together with all other possible error at the end of the run.

Comment (levitte) [inline test/dhtest.c]: Another possibility is, of course, to make sure `TEST_error` always calls `ERR_print_errors_fp(stderr)`...

Comment (pauli-oracle) [inline test/dhtest.c]: I'd take the latter option.  Any errors in the stack should be printed after the failed test.
I hadn't made it to making this change.

This means printing errors after any failed TEST_xxx conditional and TEST_error.  I'd be tempted to do it after TEST_info as well.

Comment (levitte) [inline test/dhtest.c]: I'd say that doing so in `TEST_info` would be over the top...

Comment (pauli-oracle) [inline test/dhtest.c]: WIth most TEST_infos being immediately after TEST_xxx or TEST_error, it is unlikely anything would be printed.  Still, I understand the point.  This is one that is likely better left until after we've more experience with the framework's output.

Comment (richsalz) [inline test/dhtest.c]: so i am not going to make any changes.

**Comment by richsalz** [comment]
So @levitte, you okay with this now? :)

File: test/cipherbytes_test.c [hunk 2]
```diff
@@ -19,110 +19,133 @@
...
+
+static int test_empty(void)
 {
     STACK_OF(SSL_CIPHER) *sk = NULL, *scsv = NULL;
     const unsigned char bytes[] = {0x00};
+    int ret = 0;
+
+    if (!TEST_int_eq(SSL_bytes_to_cipher_list(s, bytes, 0, 0, &sk, &scsv), 0)
+                || !TEST_ptr_null(sk)
+                || !TEST_ptr_null(scsv))
```
Comment (levitte) [inline test/cipherbytes_test.c]: 4 spaces less?

Comment (richsalz) [inline test/cipherbytes_test.c]: done; updated commit pushed.

**Comment by richsalz** [comment]
671c8cc in master, thanks.

**Pull request closed without merging**